src/clusterfuzz/_internal/bot/tasks/utasks/progression_task.py

```diff
@@ -371,26 +390,27 @@ def utask_preprocess(testcase_id, job_type, uworker_env):
   # triage cron.
   testcase.set_metadata('progression_pending', True)
   data_handler.update_testcase_comment(testcase, data_types.TaskState.STARTED)
-  progression_input = uworker_io.ProgressionTaskInput()
+  progression_input = uworker_msg_pb2.ProgressionTaskInput()
   progression_input.custom_binary = build_manager.is_custom_binary()
-  progression_input.bad_revisions.extend(build_manager.get_job_bad_revisions())
+  progression_input.bad_revisions.extend(build_manager.get_job_bad_revisions())  # pylint: disable=no-member
   # Setup testcase and its dependencies.
   setup_input = setup.preprocess_setup_testcase(testcase)
-  return uworker_io.UworkerInput(
+  testcase_proto = model._entity_to_protobuf(testcase)  # pylint: disable=protected-access
```

Comment:
Are we going to stop using Entities in preprocess/Input? Because if not, I think this is starting to become ugly without my abstractions.

Reply:
No, we will continue using entities in preprocess/input, we will have to explicitly call `_entity_to_protobuf`. At least this way, what we're doing will be explicit, instead of some magic behind the scenes. 
For what it's worth, I don't think this will pollute the code a lot.

What I suggest is we put the second commit of this PR aside until I remove the entities, then I can create a PR for the migration of all tasks to use protos and it will be clear if this is a bad idea.
@jonathanmetzman  WDYT?